main.go

```diff
@@ -277,6 +289,16 @@ func main() {
 
 	instanceName, _ := os.Hostname() // on an error, instanceName will be empty, which is ok
 
+	// A separate controller is needed because of hosted mode concerns.
+
+	nsSelReconciler := common.NamespaceSelectorReconciler{
+		Client: nsSelMgr.GetClient(),
+	}
+	if err = nsSelReconciler.SetupWithManager(nsSelMgr); err != nil {
```

Comment:
Do we need another manager? I'd prefer to consolidate with the existing manager. This will prevent your `neverEnqueue` workaround from creating an additional watch on ConfigurationPolicy objects.

Reply:
Oh, I see. Hosted mode would require a separate manager.

Reply:
And in non-hosted mode, a separate manager isn't created.